contracts/Token/RaftToken.sol

```diff
@@ -0,0 +1,77 @@
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+import { Ownable2Step } from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/// Raft Token with initial supply that be increased by up to 2% yearly after 2.5 years after deployment.
+/// Holders have the ability to burn their own tokens.
+/// Token holders have the ability to vote and participate in governance.
+/// It also supports delegating voting rights.
+contract RaftToken is Ownable2Step, ERC20Votes {
+    using SafeERC20 for IERC20;
+
+    string private _name;
+
+    string private _symbol;
+
+    /// @dev initial supply to be minted
+    uint256 public constant INITIAL_SUPPLY = 2_500_000_000e18;
+
+    /// @dev Minimum time between mints
+    uint256 public constant MIN_TIME_BETWEEN_MINTS = 365 * 1 days;
+
+    /// @dev Cap on the percentage of totalSupply that can be minted at each mint
+    uint256 public constant MINT_CAP = 10;
+
+    /// @dev The timestamp after which minting may occur
+    uint256 public immutable mintingAllowedAfter;
+
+    /// @dev The timestamp of last minting
+    uint256 public lastMintingTime;
+
+    constructor() ERC20("Raft", "RAFT") ERC20Permit("Raft") {
+        mintingAllowedAfter = block.timestamp + 2.5 * 365 * 1 days;
+        lastMintingTime = block.timestamp;
+        _mint(msg.sender, INITIAL_SUPPLY);
+    }
+
+    /// Creates `amount` new tokens for `to`.
+    /// @param account Recipient address to mint tokens to
+    /// @param amount Number of tokens to mint
+    function mint(address account, uint256 amount) external onlyOwner {
+        require(account != address(0), "Can not mint to 0x0.");
+        require(block.timestamp >= mintingAllowedAfter, "Minting not allowed yet.");
+        require(block.timestamp >= (lastMintingTime + MIN_TIME_BETWEEN_MINTS), "Not enough time between mints.");
+        require(amount <= ((MINT_CAP * totalSupply()) / 100), "Mint cap limit.");
+
+        lastMintingTime = block.timestamp;
+        _mint(account, amount);
+    }
+
+    /// Destroys `amount` tokens from the caller.
+    /// @param amount Number of tokens to burn.
+    function burn(uint256 amount) external {
+        _burn(msg.sender, amount);
+    }
+
+    function rescueTokens(IERC20 token, address to) external onlyOwner {
+        token.safeTransfer(to, token.balanceOf(address(this)));
+    }
+
+    function name() public view virtual override returns (string memory) {
+        return _name;
+    }
+
+    function symbol() public view virtual override returns (string memory) {
+        return _symbol;
+    }
+
+    function renameToken(string calldata name_, string calldata symbol_) external onlyOwner {
+        require(bytes(name_).length > 0, "Name cannot be empty.");
+        require(bytes(symbol_).length > 0, "Symbol cannot be empty.");
+        _name = name_;
+        _symbol = symbol_;
+    }
```

Comment:
I think we should emit an event here that includes the previous name and symbol and the new ones